test/cpsubsystem/AtomicLongProxyTest.js

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const { expect } = require('chai');
+const fs = require('fs');
+const Long = require('long');
+const RC = require('./../RC');
+const {
+    Client,
+    DistributedObjectDestroyedError
+} = require('../../');
+
+describe('AtomicLongProxyTest', function () {
+
+    this.timeout(30000);
+
+    let cluster;
+    let client;
+    let long;
+
+    function expectLong(expected, long) {
+        return expect(long.toString()).to.equal(Long.fromValue(expected).toString());
+    }
+
+    before(async function () {
+        cluster = await RC.createCluster(null, fs.readFileSync(__dirname + '/hazelcast_cpsubsystem.xml', 'utf8'))
+        await Promise.all([
+            RC.startMember(cluster.id),
+            RC.startMember(cluster.id),
+            RC.startMember(cluster.id)
+        ]);
+        client = await Client.newHazelcastClient({ clusterName: cluster.id });
+        long = await client.getAtomicLong('along');
+    });
+
+    afterEach(async function () {
+        // return to default value
+        await long.set(0);
+    });
+
+    after(async function () {
+        client.shutdown();
+        return RC.shutdownCluster(cluster.id);
+    });
+
+    it('should create AtomicLong with respect to given CP group', async function () {
+        const longInAnotherGroup = await client.getAtomicLong('along@mygroup');
+
+        const value1 = await longInAnotherGroup.incrementAndGet();
+        expectLong(1, value1);
+        // the following value has to be 0,
+        // as `long` belongs to the default CP group
+        const value2 = await long.get();
+        expectLong(0, value2);
+    });
+
+    it('destroy: should destroy AtomicLong', async function () {
```

Comment:
Minor: Looking at the tests at the Java side, I see that they are testing the cases of multiple destroys and getAtomicLong call after destroy. Maybe we could add those

Reply:
Makes sense. Improved in 3e4a97ecbb3e8aaf69d5e1a3fb77097cdf0f398c